csrc/scheduler/matmul.cpp

```diff
@@ -734,6 +734,21 @@ void scheduleSplitKSum(
   splitk_sum->axis(-1)->parallelize(ParallelType::Vectorize);
 }
 
+// A LoadMatrixTranspose is needed is there's a rfactor which doesn't
+// match the alloc/root domain. If there's no rfactor, then we
+// need a transpose if the alloc domain and root don't match. When we say
+// match, we refer to the innermost iter domain.
```

Comment:
I think this assumes the allocation domain of the producer matches its rfactor domain (and `out_tv`s root).